src/pytorch_lightning/strategies/colossalai.py

```diff
@@ -463,8 +463,17 @@ def broadcast(self, obj: TBroadcast, src: int = 0) -> TBroadcast:
         with _patch_cuda_is_available():
             from colossalai.communication.collective import broadcast
             from colossalai.context import ParallelMode
+            from colossalai.core import global_context as gpc
 
-        return broadcast(obj, src=src, parallel_mode=ParallelMode.GLOBAL)
+        if isinstance(obj, Tensor):
+            return broadcast(obj, src=src, parallel_mode=ParallelMode.GLOBAL)
+        else:
+            obj = [obj]
+            local_rank = gpc.get_local_rank(ParallelMode.GLOBAL)
+            if local_rank != src:
+                obj = [None]  # type: ignore[list-item]
+            torch.distributed.broadcast_object_list(obj, src, group=gpc.get_group(ParallelMode.GLOBAL))
+            return obj[0]
```

Comment:
Adding this code made `mypy` starting to complain quite a bit, please provide typing.